File: golang/first_duplicate.go
package main

func FirstDuplicateValue(array []int) int {

	// O(n) time | O(n) space
	seen := map[int]bool{}

	for _, value := range array {
		if seen[value] {
			return value
		} else {
			seen[value] = true
		}
	}
	return -1
}

File: swift/FirstDuplicate.swift
class FirstDuplicate {
    func firstDuplicateValue(_ array: inout [Int]) -> Int {
        var seen = Set<Int>()
        for value in array {
            if seen.contains(value) {
                return value
            } else {
                seen.insert(value)
            }
        }

        return -1
    }
}
